### PR TITLE
PCHR-2327: Divergence with civicrm resources in civi hr v1.6 and civi hr v1.7

### PR DESCRIPTION
**Pull request by bmnepali**
## Overview
This PR adds the fix applied in [this link](https://github.com/compucorp/civihr-employee-portal/commit/c16bbca577c72486c645f60fdff0d0588b38e2e5) by modifying the `civicrm_reource.module` file in 1.6 site. And use it to load the resources as required in appropriate section in in `civihr_employee_portal.module`.

As a result the `angular is not defined error` is also fixed in 1.6 site.

## Before
- The resources were not loaded correctly in right section causing the reqangular.min.js to load after the `ta-documents-app.js` in SSP causing angular is not defined console error as a result yoti login is also not working.
image-2017-09-28-14-55-54-980

## After
- `angular is not defined` error is not gone and SSP `/tasks-documents` has not console error.
screen shot 2017-10-02 at 3 42 27 pm

- Document modal loads correctly in ssp, without error.

-  `/welcome-page` loads with no error. After yoti login should also work as expected. Following error is because i am truing to access yoti api from my local site.

## Technical Details
1. Modified `civicrm_resource.module` file for load the reosurces in footer section in file "civicrm_resources/civicrm_resources.module".
```php
drupal_add_js(ltrim(str_replace(DRUPAL_ROOT, '', $js_file), '/'),  ['scope' => 'footer']);
```

2. Modify `_add_script_to_page()` to receive `jsOptions` param which will be used to load the resources in specific sections (`header`, `body` and `footer`)
```php
function _add_script_to_page($resources, $extension = NULL, $page = NULL, $scope = 'footer') {
    $options = ['type' => 'file', 'scope' => $scope];
    ...
    else {
       if (is_array($resources)) {
          foreach ($resources as $value) {
             drupal_add_js(drupal_get_path('module', 'civihr_employee_portal') . $value, $options);
          }
       }
       else {
          drupal_add_js(drupal_get_path('module', 'civihr_employee_portal') . $resources, $options);
       }
    }
}
```

3. Load the resources in order in SSP in file "civihr_employee_portal/civihr_employee_portal.module"
```php
_add_script_to_page(['/js/tasks.js'], NULL, 'tasks-and-documents', 'header');
 _add_script_to_page(['/js/ta-documents-app.js'], NULL, 'tasks-and-documents');
_add_script_to_page(['tasks-assignments.min.js'], 'uk.co.compucorp.civicrm.tasksassignments', 'tasks-and-documents');
```

## Comments
The same fixes are added in 1.7-wip branch and tested, and it fixes the reported issue for 1.7 site as well.

### CODE REVIEW

File: civihr_employee_portal/civihr_employee_portal.module [hunk 1]
```diff
@@ -135,8 +135,9 @@ function _setup_modals() {
  * @param string $resources the name of the script file (extension included)
  * @param string $extension
  * @param string $page
+ * @param array $jsOptions
  */
-function _add_script_to_page($resources, $extension = NULL, $page = NULL) {
+function _add_script_to_page($resources, $extension = NULL, $page = NULL, $jsOptions = ['scope' => 'footer']) {
```
Comment (AkA84) [inline civihr_employee_portal/civihr_employee_portal.module]: does the additional param need to be an array at all? can't it be just `$scope = 'footer'`?

Comment (mickadoo) [inline civihr_employee_portal/civihr_employee_portal.module]: Is 'footer' a reasonable deafult, i.e. would you assume most scripts should be added to the footer?

Comment (AkA84) [reply civihr_employee_portal/civihr_employee_portal.module]: @mickadoo yeah they should be

Comment (bmnepali) [reply civihr_employee_portal/civihr_employee_portal.module]: yes! fixed it.

Comment (bmnepali) [reply civihr_employee_portal/civihr_employee_portal.module]: yes!

File: civicrm_resources/civicrm_resources.module [hunk 2]
```diff
@@ -55,7 +55,7 @@ function civicrm_resources_add_resources($extention_path, $req_files = []) {
     drupal_add_css(ltrim(str_replace(DRUPAL_ROOT, '', $css_file), '/'));
   }
   foreach ($js_list as $key => $js_file) {
-    drupal_add_js(ltrim(str_replace(DRUPAL_ROOT, '', $js_file), '/'));
+    drupal_add_js(ltrim(str_replace(DRUPAL_ROOT, '', $js_file), '/'),  array('scope' => 'footer'));
```
[the remark this answers is not in the document]
Comment (bmnepali) [reply civicrm_resources/civicrm_resources.module]: Right! will fix it

File: civihr_employee_portal/civihr_employee_portal.module [hunk 3]
```diff
@@ -135,8 +135,10 @@ function _setup_modals() {
  * @param string $resources the name of the script file (extension included)
  * @param string $extension
  * @param string $page
+ * @param array $jsOptions
```
Comment (davialexandre) [inline civihr_employee_portal/civihr_employee_portal.module]: Since you've changed the params, you need to update the docblock as well

Comment (bmnepali) [reply civihr_employee_portal/civihr_employee_portal.module]: fixed!

File: civicrm_resources/civicrm_resources.module (hunk 2 above)
[the remark this answers is not in the document]
Comment (davialexandre) [reply civicrm_resources/civicrm_resources.module]: fair enough

**Pull request merged**